### PR TITLE
CTCTRADERS-2174: Special mentions xml

### PR DESCRIPTION
**Pull request by MatthewAlexanderBeck**

### CODE REVIEW

File: app/services/DeclarationRequestService.scala [hunk 1]
```diff
@@ -124,10 +124,12 @@ class DeclarationRequestService @Inject()(
             previousAdministrativeReferences = previousAdministrativeReference(itemSection.previousReferences),
             producedDocuments                = producedDocuments(itemSection.producedDocuments),
             specialMention = {
+              val specialMentions = itemSection.specialMentions.map(SpecialMentionConversion).getOrElse(Seq.empty)
```
Comment (amaalali) [inline app/services/DeclarationRequestService.scala:127]: Test

File: app/services/DeclarationRequestService.scala [hunk 2]
```diff
@@ -124,10 +124,12 @@ class DeclarationRequestService @Inject()(
...
+
               if (index == 0) {
-                GuaranteeLiabilityAmountConversion(guaranteeDetails)
+                SpecialMentionGuaranteeLiabilityConversion(guaranteeDetails) ++ specialMentions
               } else {
-                Seq.empty // TODO add users special mentions here and above
+                specialMentions
```
Comment (amaalali) [inline app/services/DeclarationRequestService.scala:132]: As a refactor, would it make sense to move this conversion block for Special Mentions into a separate class, similar to what we did for the conversion of guarantees to special mentions?

File: app/services/SpecialMentionConversion.scala [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+package services
+
+import cats.data.NonEmptyList
+import models.Convert
+import models.journeyDomain.SpecialMentionDomain
+import models.messages.goodsitem.{SpecialMention, SpecialMentionExportFromGB, SpecialMentionNoCountry}
+
+object SpecialMentionConversion extends Convert[NonEmptyList, Seq, SpecialMentionDomain, SpecialMention] {
```
Comment (amaalali) [inline app/services/SpecialMentionConversion.scala:24]: Can we make this package private since we don't need it outside of services?

**Review by amaalali — APPROVED** [review]

File: app/services/DeclarationRequestService.scala (hunk 2 above)
Comment (MatthewAlexanderBeck) [reply app/services/DeclarationRequestService.scala:132]: Will follow up this suggestion with a refactor PR, cheers.

File: app/services/SpecialMentionConversion.scala (hunk 3 above)
Comment (MatthewAlexanderBeck) [reply app/services/SpecialMentionConversion.scala:24]: Will cover with above PR

**Pull request merged**